pkg/docker/config/config.go

```diff
@@ -338,17 +363,21 @@ func RemoveAuthentication(sys *types.SystemContext, registry string) error {
 	isLoggedIn := false
 
 	removeFromCredHelper := func(helper string) {
-		err := deleteAuthFromCredHelper(helper, registry)
-		if err == nil {
-			logrus.Debugf("Credentials for %q were deleted from credential helper %s", registry, helper)
-			isLoggedIn = true
-			return
-		}
-		if credentials.IsErrCredentialsNotFoundMessage(err.Error()) {
-			logrus.Debugf("Not logged in to %s with credential helper %s", registry, helper)
-			return
+		if isNamespaced {
+			logrus.Debugf("Not removing credentials: %v", unsupportedNamespaceErr(helper))
```

Comment:
Creating and parsing an `error` object is a rather expensive way to construct a message, and more importantly I can’t see any benefit: it just makes it harder to see that the debug line in practice explains the reason.